### PR TITLE
[FIX][Performance] product: recompute fields of all new variants in one time

### PR DESCRIPTION
**Pull request by fmdl**
Description of the issue/feature this PR addresses:
Before this PR, when you create 4000 variantes, the times to create each variante is very very very long. Because `product.template` and `product.product` fields are recomputed each time.

before 20 min, after 4 min.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fmdl** [comment]
@jco-odoo

**Comment by jco-odoo** [comment]
I like your idea.  Any idea of the perf you gain and the fields that took the most time? I suppose all that depends on product_variant_ids like sale_count, ...

File: addons/product/models/product_template.py [hunk 1]
```diff
@@ -442,4 +443,5 @@ def create_variant_ids(self):
                 except (psycopg2.Error, except_orm):
                     variant.write({'active': False})
                     pass
```
Comment (jco-odoo) [inline addons/product/models/product_template.py]: we might remove this pass

File: addons/product/models/product_template.py [hunk 2]
```diff
@@ -391,6 +391,7 @@ def _price_get(self, products, ptype='list_price'):
     def create_variant_ids(self):
         Product = self.env["product.product"]
         AttributeValues = self.env['product.attribute.value']
+        new_products = Product
```
Comment (jco-odoo) [inline addons/product/models/product_template.py]: Maybe with_context before instead of each time?

**Comment by fmdl** [comment]
@jco-doo, it don't time, but is some things like that : before 20 min, after 4 min.

File: addons/product/models/product_template.py [hunk 3]
```diff
@@ -428,7 +429,7 @@ def create_variant_ids(self):
 
             # create new product
             for variant_ids in to_create_variants:
-                new_variant = Product.create({
+                new_products |= Product.create({
```
Comment (xmo-odoo) [inline addons/product/models/product_template.py]: That's unnecessary, `recompute()` just triggers an execution of all pending stored function field computations, it doesn't care about the recorset (it doesn't even care about the model it's called on, that's just an API convenience).

**Comment by fmdl** [comment]
@jco-odoo @xmo-odoo , I close because it can solve with the new create_multi

**Pull request closed without merging**